compiler/passes/resolveIntents.cpp

```diff
@@ -100,15 +108,16 @@ static IntentTag blankIntentForThisArg(Type* t) {
 }
 
 void resolveArgIntent(ArgSymbol* arg) {
-  arg->intent = 
-    arg->hasFlag(FLAG_ARG_THIS) && arg->intent == INTENT_BLANK ?
-    blankIntentForThisArg(arg->type) :
-    concreteIntent(arg->intent, arg->type);
+  if (arg->hasFlag(FLAG_ARG_THIS) && arg->intent == INTENT_BLANK)
+    arg->intent = blankIntentForThisArg(arg->type);
+  else
+    arg->intent = concreteIntent(arg->intent, arg->type);
```

Comment:
Wow, I don't think I've seen a ternary operator that nasty. I'm glad you replaced it with a conditional.